pkgs/by-name/kc/kclvm/package.nix

```diff
@@ -7,27 +7,26 @@
 }:
 rustPlatform.buildRustPackage rec {
   pname = "kclvm";
-  version = "0.8.7";
+  version = "0.9.0";
 
   src = fetchFromGitHub {
     owner = "kcl-lang";
     repo = "kcl";
     rev = "v${version}";
-    hash = "sha256-ieGpuNkzT6AODZYUcEanb7Jpb+PXclnQ9KkdmlehK0o=";
+    hash = "sha256-xVKoH0wu8QP4g/klTidx1wdt/oFoTX5GS/J9uVGgKjo=";
   };
 
   sourceRoot = "source/kclvm";
   cargoLock = {
     lockFile = ./Cargo.lock;
     outputHashes = {
       "inkwell-0.2.0" = "sha256-JxSlhShb3JPhsXK8nGFi2uGPp8XqZUSiqniLBrhr+sM=";
+      "protoc-bin-vendored-3.1.0" = "sha256-RRqpPMJygpKGG5NYzD93iy4htpVqFhYMmfPgbRtpUqg=";
     };
   };
 
   nativeBuildInputs = [ pkg-config protobuf ];
 
-  patches = [ ./enable_protoc_env.patch ];
-
```

Comment:
Is the patch no longer needed ?
I updated it on my branch. https://github.com/NixOS/nixpkgs/compare/master...YvanDaSilva:nixpkgs:kcl-update

Reply:
@selfuryon @Peefy 
Feel free to have a look, I updated https://github.com/NixOS/nixpkgs/compare/master...YvanDaSilva:nixpkgs:kcl-update

But I don't think it does what it is supposed to do. Check the slack thread for more info.
TLDR: It produces a kcl file of 453B, something is borked.
kclvm and kclvm_cli seem ok though.
It's the compilation of the go program that is not working as intended, I had to:
```
    export HOME=$(mktemp -d)
```

in post install. Otherwise it would panic:
```


goroutine 1 [running]:

kcl-lang.io/cli/cmd/kcl/commands.NewModCmd()

        kcl-lang.io/cli/cmd/kcl/commands/mod.go:49 +0x350

kcl-lang.io/cli/cmd/kcl/commands.New()

        kcl-lang.io/cli/cmd/kcl/commands/root.go:119 +0x238

main.main()

        kcl-lang.io/cli/cmd/kcl/main.go:14 +0x1c

panic: failed to load config file '' for kpm.

internal bugs, failed to load working directory

could not access $KCL_PKG_PATH.

internal bugs, failed to create directory

mkdir /homeless-shelter: read-only file system

```

Reply:
Made the PR HERE: https://github.com/NixOS/nixpkgs/pull/327390

It works on both nixos x86 and darwin aarch